### PR TITLE
Setting type attribue with setAttribute function

### PR DESCRIPTION
**Pull request by Rafi993**
This is done to prevent crash in IE11 as mentioned in this issue https://github.com/preactjs/preact/issues/2110

### CODE REVIEW

**Comment by Rafi993** [comment]
> we usually add a test to assert if this is done correctly to prevent future regressions

sorry. I'll add test for this too.

**Comment by JoviDeCroock** [comment]
No need to say sorry, just notifying thanks for tackling this

**Comment by marvinhagemeister** [comment]
That's awesome, thank you so much for the PR :+1: :tada: For tests you can take a look at the existing ones. We have a few where we check if an operation throws an error via `expect(() => ...).to.not.throw` or something similar :+1:

File: test/browser/render.test.js [hunk 1]
```diff
@@ -91,6 +91,13 @@ describe('render()', () => {
 		expect(scratch.childNodes[0].nodeName).to.equal('SPAN');
 	});
 
+	it('should create input tag with type date', () => {
+		render(<input type="date" />, scratch);
+		expect(scratch.childNodes).to.have.length(1);
+		expect(scratch.childNodes[0].nodeName).to.equal('INPUT');
+		expect(scratch.childNodes[0].type).to.equal('date');
```
Comment (marvinhagemeister) [inline test/browser/render.test.js]: I think we should test something that more closely resembles the original bug report. The fix in this PR specifically addresses the case where setting `input.date` would throw an error in IE11. The test case should reflect only that. On our CI our test suite will be run in various browsers, including IE11. So we only need to check if it throws or not.

```js
it('should not throw in IE11 with type date', () => {
  expect(() => render(<input type="date" />, scratch)).to.not.throw();
});
```

**Comment by Rafi993** [comment]
Thanks for the help @marvinhagemeister  @JoviDeCroock you guys are awesome !!!

**Pull request merged**